twitter_scrooge/twitter_scrooge.bzl

```diff
@@ -73,13 +75,12 @@ def _collect_owned_srcs(targets):
 def _collect_external_jars(targets):
   r = []
   for target in targets:
-    if hasattr(target, "thrift"):
-      thrift = target.thrift
-      if hasattr(thrift, "external_jars"):
-        for jar in thrift.external_jars:
-          r.append(depset(_jar_filetype.filter(jar.files)))
-      r.append(depset(_jar_filetype.filter(thrift.transitive_external_jars)))
-  return depset(transitive = r)
+    if ThriftInfo in target:
+      thrift = target[ThriftInfo]
+      for jar in thrift.external_jars:
+        r.extend(_jar_filetype.filter(jar.files))
+      r.extend(_jar_filetype.filter(thrift.transitive_external_jars))
+  return depset(r)
```

Comment:
consider using the transitive parameter on the depset constructor. Something like this might work:
```
depset(transitive =
  [_jar_filetype.filter(jar.files) for jar in thrift.external_jars] +
  [_jar_filetype.filter(thrift.transitive_external_jars)]
)
```

Reply:
Also note that `FileType` is deprecated and we should move away from it (maybe in another PR).

Reply:
I don't think that works because transitive is a list of depsets while the arg to depset is a list of items (non-depset). I could make singleton depsets, but I assume that is slower than just putting it in the first arg since it is adding more nesting.

Reply:
👍 makes sense. I'm not terribly familiar with the thrift code so it wasn't immediately apparent what's a depset and what's not.

Reply:
yeah, me neither. Especially due to the lack of types... :(